src/types/index.ts

```diff
@@ -407,25 +290,88 @@ export interface Tag {
   id: string;
   isOfficial: boolean;
 }
-
 export interface ReplComment {
   id: number;
   body: string;
   user: User;
 }
 
-// Response types
+/*****************************************************************
+ * * Miscalleneous / React Types
+ *****************************************************************/
 
-export type GraphResponse<T> = Promise<T | never>;
+export enum HandshakeStatus {
+  Ready = "ready",
+  Error = "error",
+  Loading = "loading",
+}
 
-export type ReplQueryOutput = GraphResponse<{
-  repl: Repl;
-}>;
+export type StrError = { error: string };
+export type NullableStrError = { error: string | null };
+export type VoidPromise = Promise<void>;
+export type VoidFunction = () => void;
 
-export type UserByUsernameQueryOutput = GraphResponse<{
-  userByUsername: User;
-}>;
+/*****************************************************************
+ * * Extension Port Wrapper
+ *****************************************************************/
 
-export type UserQueryOutput = GraphResponse<{
-  user: User;
-}>;
+export type ExtensionPortAPI = {
+  handshake: () => void;
+
+  // fs Module
+  readFile: (
+    path: string,
+    encoding: "utf8" | "binary" | null
+  ) => Promise<{ content: string } | StrError>;
+  writeFile: (
+    path: string,
+    content: string | Blob
+  ) => Promise<{ success: boolean } | StrError>;
+  readDir: (path: string) => Promise<{
+    children: Array<DirectoryChildNode>;
+    error: string;
+  }>;
+  createDir: (path: string) => Promise<{} | StrError>;
+  deleteFile: (path: string) => Promise<{} | StrError>;
+  deleteDir: (path: string) => Promise<{} | StrError>;
+  move: (path: string, to: string) => Promise<NullableStrError>;
+  copyFile: (path: string, to: string) => Promise<NullableStrError>;
+  watchFile: (path: string, watcher: WatchFileWatchers) => VoidFunction;
+  watchTextFile: (path: string, watcher: WatchTextFileWatchers) => VoidFunction;
```

Comment:
we should probably make our result types more consistent

Reply:
This will require some changes in replit-web, right?
Should I keep/revert to original for now?

Reply:
nah i think this is fine

Reply:
we'll simplify both